soweego/linker/train.py

```diff
@@ -18,38 +18,63 @@
 import pandas as pd
 from pandas import MultiIndex, concat
 from sklearn.externals import joblib
+from sklearn.model_selection import GridSearchCV
 
-from soweego.commons import constants, data_gathering, target_database
+from soweego.commons import constants, data_gathering, target_database, utils
 from soweego.linker import blocking, workflow
 
 LOGGER = logging.getLogger(__name__)
 
 
-@click.command()
+@click.command(context_settings={'ignore_unknown_options': True, 'allow_extra_args': True})
 @click.argument('classifier', type=click.Choice(constants.CLASSIFIERS))
 @click.argument('target', type=click.Choice(target_database.available_targets()))
 @click.argument('target_type', type=click.Choice(target_database.available_types()))
-@click.option('-b', '--binarize', default=0.1, help="Default: 0.1")
+@click.option('--tune', is_flag=True, help='Run grid search for hyperparameters tuning. Default: no.')
+@click.option('-k', '--k-folds', default=5, help="Number of folds for hyperparameters tuning. Implies '--tune' Default: 5.")
 @click.option('-d', '--dir-io', type=click.Path(file_okay=False), default=constants.SHARED_FOLDER,
               help="Input/output directory, default: '%s'." % constants.SHARED_FOLDER)
-def cli(classifier, target, target_type, binarize, dir_io):
+@click.pass_context
+def cli(ctx, classifier, target, target_type, tune, k_folds, dir_io):
     """Train a probabilistic linker."""
+    kwargs = utils.handle_extra_cli_args(ctx.args)
+    if kwargs is None:
+        return 1
 
-    model = execute(
-        constants.CLASSIFIERS[classifier], target, target_type, binarize, dir_io)
+    model = execute(constants.CLASSIFIERS[classifier],
+                    target, target_type, tune, k_folds, dir_io, **kwargs)
     outfile = os.path.join(
         dir_io, constants.LINKER_MODEL % (target, target_type, classifier))
     joblib.dump(model, outfile)
     LOGGER.info("%s model dumped to '%s'", classifier, outfile)
 
 
-def execute(classifier, catalog, entity, binarize, dir_io):
-    goal = 'training'
+def execute(classifier, catalog, entity, tune, k, dir_io, **kwargs):
+    if tune and classifier is constants.SINGLE_LAYER_PERCEPTRON:
```

Comment:
Also here, after merging with master the `mlp` classifier should also be considered 

```suggestion
    if tune and classifier in (constants.SINGLE_LAYER_PERCEPTRON, 
                      constants.MULTILAYER_CLASSIFIER):
```